crates/meilisearch-auth/src/store.rs

```diff
@@ -345,3 +288,32 @@ where
     let head = head.try_into().ok()?;
     Some((head, tail))
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct KeyMasks {
+    bitflags: u32, // Bitflags for bit_maks
+    indexes: Vec<IndexUidPattern>,
+    expires_at: Option<OffsetDateTime>,
+}
+
+impl KeyMasks {
+    fn has_action(&self, bitflags: u32) -> bool {
+        for (index, action_flag) in enum_iterator::all::<Action>().enumerate() {
+            if action_flag.bits() == bitflags && (self.bitflags & (1 << index) != 0) {
+                return true;
+            }
+        }
+        false
+    }
```

Comment:
To remove in favor of a bitflag comparison:
```suggestion
    fn is_bitflag_authorized(&self, bitflags: u32) -> bool {
        bitflags.difference(self.bitflags).is_empty()
    }
```

Reply:
there is no method `difference` in `u32`

Reply:
Not sure, but wouldn't `bitflags & self.bitflags == 0` works?

Reply:
huhh i missed it thanks

Reply:
I think then we need to import `bitflags` crate here if we want to use Flags trait or should I implement something like this??

Reply:
It's already imported, it's used for the `Actions`, when doing `action_flag.bits()`, you are using the `Flags` trait.

Maybe, we should use the type Action everywhere, because it seems that it can contain "unnamed" flags that could correspond to the union of several Actions.

```rust
    fn is_bitflag_authorized(&self, bitflags: Action) -> bool {
        // self.bitflag has type: Action
        bitflags.difference(self.bitflags).is_empty()
    }
```

this may work 🤔

Reply:
Hey 
```rust
fn is_bitflag_authorized(&self, bitflags: u32) -> bool {
        Action::from_bits(bitflags)
            .map(|flags| flags.difference(Action::from_bits(self.bitflags).unwrap_or(Action::empty())).is_empty())
            .unwrap_or(false)
    }
```
I have tried this and I think this is good to go and we don't need to change type of bitflags to `Action` 
I have pushed it tell me if it seems wrong